### ISSUE TITLE
Displayed item's buy/sell price not updated when using DynamicPriceProvider.

### ISSUE THREAD

**Issue by astetyn** [pablo67340/GUIShop#94]
When I buy an item, my dynamic price provider changes the price but item has still old value in his lore. So when I click once more, I pay different price than displayed. The price is also not updated in category inventory.

Spigot version: 1.16.5
Plugin version: 8.5

![Screenshot_1](https://user-images.githubusercontent.com/39633703/114276312-5c207d80-9a26-11eb-8059-873387bba774.png)

**Comment by pablo67340** [comment]
I will look into this and see if I can find the issue. Would you mind adding me on Discord? I may need the plugin you are working on to find the exact cause of the issue.

Discord: Bryce#0001

**Comment by pablo67340** [comment]
Perhaps @A248 can look into this when he has time?

**Comment by A248** [comment]
For testing purposes, having the plugin you use could really help with this as I never finished implementing and testing my own DynamicPriceProvider. That would save us some time attempting to recreate the same environment.

**Comment by astetyn** [comment]
I tested this with older version of GUIShop (with my changes included) and it was working. The only case where problem appeared was when player A bought an item and player B had open shop at that time - the price did not update obviously for player B.
In zip I included an example image of how price should change over time. Exact numbers depends on buy/sell price set in config (it is calculated from them).
I don't have much time testing this for newer version, but this dynamic price addon should remain the same because interface does not change.
[guishopaddon.zip](https://github.com/pablo67340/GUIShop/files/6582843/guishopaddon.zip)
